packages/server/lib/middleware/resource-capping.middleware.ts

```diff
@@ -0,0 +1,37 @@
+import type { Request, Response, NextFunction } from 'express';
+import {
+    accountService,
+    getEnvironmentAndAccountId,
+    errorManager,
+    connectionCreationStartCapCheck as connectionCreationStartCapCheckHook
+} from '@nangohq/shared';
+
+export const authCheck = async (req: Request, res: Response, next: NextFunction) => {
+    const { success, error, response } = await getEnvironmentAndAccountId(res, req);
+
+    if (!success || response === null) {
+        errorManager.errResFromNangoErr(res, error);
+        return;
+    }
+
+    const { accountId, environmentId } = response;
+
+    const account = await accountService.getAccountById(accountId);
+
+    if (!account) {
+        errorManager.errRes(res, 'unknown_account');
+        return;
+    }
+
+    const { providerConfigKey } = req.params;
+
+    if (account.is_capped && providerConfigKey) {
+        const isCapped = await connectionCreationStartCapCheckHook({ providerConfigKey, environmentId });
+        if (isCapped) {
+            errorManager.errRes(res, 'resource_capped');
+            return;
+        }
+    }
+
+    next();
+};
```

Comment:
No, it will just get executed for all calls that are creating a connection using their public key auth. 

```
app.route('/oauth/connect/:providerConfigKey').get(apiPublicAuth, oauthController.oauthRequest.bind(oauthController));
app.route('/oauth2/auth/:providerConfigKey').post(apiPublicAuth, oauthController.oauth2RequestCC.bind(oauthController));
app.route('/api-auth/api-key/:providerConfigKey').post(apiPublicAuth, apiAuthController.apiKey.bind(apiAuthController));
app.route('/api-auth/basic/:providerConfigKey').post(apiPublicAuth, apiAuthController.basic.bind(apiAuthController));
app.route('/app-store-auth/:providerConfigKey').post(apiPublicAuth, appStoreAuthController.auth.bind(appStoreAuthController));
app.route('/unauth/:providerConfigKey').post(apiPublicAuth, unAuthController.create.bind(unAuthController));
```